src/handle.rs

```diff
@@ -0,0 +1,36 @@
+#![allow(dead_code)]
```

Comment:
Why allow dead code? If `as_ref` and `as_mut` are not needed, they're not needed so I'd remove them. They can be added later when needed.

Reply:
`sendqueue.rs` use them, I could change the code of sendqueue or add cfg feature gate just for them, I didn't like any of the solution.

Reply:
Ah okay, leave a comment explaining this then.